android/src/main/kotlin/com/situm/flutter/ar/situm_ar/scene/ARQuality.kt

```diff
@@ -0,0 +1,287 @@
+    private var arDisplacementConf: Double = 0.0
+    private var situmDisplacementConf: Double = 0.0
+
+    private var arLocationBuffer: MutableList<LocationCoordinates> = mutableListOf()
+    private var situmLocationBuffer: MutableList<LocationCoordinates> = mutableListOf()
+
+
+    fun updateARLocation(worldPosition: Position, worldRotation: Rotation) {
+        arLocationBuffer.add(
+            LocationCoordinates(
+                worldPosition.x.toDouble(),
+                worldPosition.z.toDouble(), worldRotation.y.toDouble(), System.currentTimeMillis()
+            )
+        )  //TODO: Check rotation
+        if (arLocationBuffer.size > BUFFER_SIZE) {
+            arLocationBuffer.removeAt(0)
+        }
+    }
+
+    fun updateSitumLocation(location: Location) {
+        if (situmLocationBuffer.isNotEmpty() && situmLocationBuffer.last().floorIdentifier != location.floorIdentifier) {
+            situmLocationBuffer.clear()
+            arLocationBuffer.clear()
+            resetThreshold()
+        }
+        situmLocationBuffer.add(
+            LocationCoordinates(
+                location.cartesianCoordinate.x,
+                location.cartesianCoordinate.y,
+                location.cartesianBearing.degrees(),
+                System.currentTimeMillis(),
+                location.floorIdentifier,
+                location.accuracy.toLong(),
+                location.hasBearing()
+            )
+        )
+        if (situmLocationBuffer.size > BUFFER_SIZE) {
+            situmLocationBuffer.removeAt(0)
+        }
+    }
+
+
+    fun hasToResetWorld(): Boolean {
+        logExecutionTime(" >> Update Confidence ") {
+            updateConfidence()
+        }
+        return checkIfHasToRefreshAndUpdateThreshold(quality, arConf, situmConf)
+    }
+
+    fun updateConfidence() {
+        if (situmLocationBuffer.isEmpty() || arLocationBuffer.isEmpty()) {
+            arConf = 0.0
+            situmConf = 0.0
+            quality = 0.0
+            return
+        }
+        val totalDisplacementSitum = computeTotalDisplacement(situmLocationBuffer)
+        val totalDisplacementAR = computeTotalDisplacement(arLocationBuffer)
+
+        val odometriesDistance = estimateOdometriesMatch(arLocationBuffer, situmLocationBuffer)
+
+        situmDisplacementConf = totalDisplacementConf(totalDisplacementSitum)
+        arDisplacementConf = totalDisplacementConf(totalDisplacementAR)
+        arConf = estimateArConf()
+        situmConf = estimateSitumConf()
+        odometriesDistanceConf = odometriesDifferenceConf(odometriesDistance)
+
+        quality =
+            situmDisplacementConf * arDisplacementConf * arConf * situmConf * odometriesDistanceConf
+    }
+
+
+    private fun estimateOdometriesMatch(
+        arLocationBuffer: MutableList<LocationCoordinates>,
+        situmLocationBuffer: MutableList<LocationCoordinates>
+    ): Double {
+        var transformedARTrajectory = transformTrajectory(arLocationBuffer)
```

Comment:
Revisa o lint porfa, parece que hai algúns warnings. Este por exemplo é leve (usar val en vez de var), pero hai máis por aí abaixo.